src/Core/Config/Adapter/IConfigAdapter.php

```diff
@@ -13,52 +13,59 @@ interface IConfigAdapter
 	 *
 	 * @param string  $cat The category of the configuration values to load
 	 *
-	 * @return void
+	 * @return array
 	 */
 	public function load($cat = "config");
 
 	/**
 	 * Get a particular user's config variable given the category name
 	 * ($family) and a key.
 	 *
-	 * @param string  $cat           The category of the configuration value
-	 * @param string  $k             The configuration key to query
-	 * @param mixed   $default_value optional, The value to return if key is not set (default: null)
-	 * @param boolean $refresh       optional, If true the config is loaded from the db and not from the cache (default: false)
+	 * @param string  $cat The category of the configuration value
+	 * @param string  $key The configuration key to query
 	 *
-	 * @return mixed Stored value or null if it does not exist
+	 * @return mixed Stored value or "!<unset>!" if it does not exist
 	 */
-	public function get($cat, $k, $default_value = null, $refresh = false);
```

Comment:
Why did you remove the default value here? Are default values still handled at all?

Reply:
The `Adapter` and `ConfigCache` aren't responsible for default values anymore. The adapter is for `DBA` backend loadings (JIT or complete) and the `ConfigCache` is responsible for caching config values. The `Configuration` class is the glue between them and is the facade for the outer world (which contains the default  values).

My overall target was to clean up the classes to be small, readable and testable.

Reply:
f.e. (just an idea) we now can easily create a `CacheConfigAdapter` which initially loads the whole config into a key-value storage (f.e. redis) and sync the result in a specific intervall back to the db. It has no impact on the usage of the Config, just another adapter :)

Btw. this would maybe something to rapidly decrease the db-calls for config keys